dm/common.go

```diff
@@ -1,111 +1,101 @@
 package dm
 
-type Attribute struct {
-	Name      string `json:"name"`
-	Region	  string `json: region"`
-	Extension struct {
-		Data    map[string]interface{} `json:"data"`
-		Version string                 `json:"version"`
-		Type    string                 `json:"type"`
-		Schema  struct {
-			Href string `json:"href"`
-		} `json:"schema"`
-	} `json:"extension"`
+type ForgeResponseObject struct {
+	JsonApi 				JsonAPI 		`json:"jsonApi`
+	Links 					Links 			`json:"links"`
+	Data 					Data 			`json:"data"`
+	Included 				*[]Data 		`json:"included, omitempty"`
 }
 
-type Content struct {
-	Relationships 	Relationships 	`json:"relationships, omitempty"`
-	Attributes 	  	Attribute 		`json:"attributes, omitempty"`
-	Type       		string    		`json:"type, omitempty"`
-	Id         		string    		`json:"id, omitempty"`
-	Links      		Link      		`json:"links, omitempty"`
+type ForgeResponseArray struct {
+	JsonApi 				JsonAPI 		`json:"jsonApi`
+	Links 					Links 			`json:"links"`
+	Data 					[]Data 			`json:"data"`
+	Included 				*[]Data 		`json:"included, omitempty"`
 }
 
-type DataDetails struct {
-	Data    	[]Content 	`json:"data, omitempty"`
-	JsonApi 	JsonAPI   	`json:"jsonapi, omitempty"`
-	Links   	Link      	`json:"links, omitempty"`
-}
-
-type FolderContents struct {
-	JsonApi 	JsonAPI   	`json:"jsonapi, omitempty"`
-	Links   	Link      	`json:"links, omitempty"`
-	Data    	[]Content 	`json:"data, omitempty"`
-	Included 	[]Content 	`json:"included, omitempty"`
+type JsonAPI struct {
+	Version 				string 			`json:"version"`
 }
 
-type ItemDetails struct {
-	Data    	Content 	`json:"data, omitempty"`
-	JsonApi 	JsonAPI   	`json:"jsonapi, omitempty"`
-	Links   	Link      	`json:"links, omitempty"`
-	Included 	[]Content 	`json:"included, omitempty"`
+type Links struct {
+	Self 					*Href 			`json:"self, omitempty"`
+	Related 				*Href 			`json:"related, omitempty"`
+	First 					*Href 			`json:"first, omitempty"`
+	Prev 					*Href 			`json:"prev, omitempty"`
+	Next 					*Href 			`json:"next, omitempty"`
 }
 
-type Hub struct {
-	Links 		Link 		`json:"links, omitempty"`
-	Data 		[]Content 	`json:"data, omitempty"`
+type Data struct {
+	Type 					string 			`json:"type"` 	 
+	Id 					string 			`json:"id"`
+	Attributes 				*Attributes 		`json:"attributes, omitempty"`
+	Relationships 				*Relationships 		`json:"relationships, omitempty"`
+	Links 					*Links 			`json:"links, omitempty"`
 }
 
-type JsonAPI struct {
-	Version 	string 		`json:"version, omitempty"`
+type Attributes struct {
+	Name      				string 			`json:"name"`
+	Extension 				Extension 		`json:"extension"`
+	Region	  				*string 		`json:"region, omitempty"`
```

Comment:
curious what makes this one a `*string` and `Name` above a `string`?

Reply:
The fields with a pointer relate to the `optional` or "omitempty" nature of them. Based on my understanding from [this article](https://www.sohamkamani.com/blog/golang/2018-07-19-golang-omitempty/), the pointer allows the existence of a `null` value for a given field.

Reply:
ooo cool cool
I see, so name isn't optional so its not a pointer?